File: 01_Language/01_Functions/python/html_entity_decode.py
# coding: utf-8

import html


def html_entity_decode(string):
    return html.unescape(string)


if __name__ == '__main__':
    print(html_entity_decode("I'll &quot;walk&quot; the &lt;b&gt;dog&lt;/b&gt; now"))
    print(html_entity_decode("&lt;Il &eacute;tait une fois un &ecirc;tre&gt;."))

File: 01_Language/01_Functions/php/html_entity_decode.php
<?php

$orig = "I'll \"walk\" the <b>dog</b> now";

$a = htmlentities($orig, ENT_QUOTES | ENT_HTML401);

$b = html_entity_decode($a);

echo $a; // I'll &quot;walk&quot; the &lt;b&gt;dog&lt;/b&gt; now
echo "\n";
echo $b; // I'll "walk" the <b>dog</b> now
echo "\n";
